### ISSUE TITLE
Unexpected behavior of request.params with semicolon in the body

### ISSUE THREAD

**Issue by vikas-nexcom** [Pylons/pyramid#3672]
Hi everyone, 

I am new to Pyramid and trying to address a bug that I observed in the request. params functionality. 

**Details**
Python: `3.8.5`
Pyramid version: `1.10.4`
Frontend: `Angular 11`

![Screenshot from 2021-06-18 16-17-48](https://user-images.githubusercontent.com/74184952/122575711-4f855a80-d051-11eb-8ea1-7cf6f3062a84.png)

I have noticed that if I include `;` in the body.[ `request.parms`](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/webob.html#request-json-body) unexpectedly delimits the dictionary eventually leading to :
`
json.decoder.JSONDecodeError: Unterminated string starting at: line 1 column 80 (char 79)
`
Here is what happens if I print `request.params`:
```
# if a semicolon is present in the body
"entryType":"something","body":"this'), ('","isComplete":false

# if a semicolon is not present in the body (works fine)
"entryType":"something","body":"this","isComplete":false
```

There is no problem within `request.body`, which makes certain that the problem lies within `request.params`

```
[{
    "id": 0,
    "mid": "",
    "userId": "user",
    "fullName": "",
    "entryType": "something",
    "body": "this;",
    "isComplete": false,
    "inPerson": false,
    "planType": "Challenge",
    "cAtId": "",
    "courseName": "",
    "datetimeCreated": "6/15/2021, 09:00 PM",
    "dateTimeModified": "",
    "dueDate": "",
    "userTz": "",
    "canEdit": true,
    "tId": 0,
    "programId": 0,
    "someReplies": []
}]
```
If anyone can share any insights, that will be really great. If you like more information, please let me know. 

Thanks,

**Comment by stevepiercy** [comment]
Please include an example of the request object that you send to your Pyramid app. I am not convinced this is a bug in Pyramid or its dependency Webob, as I expect that an issue with a semicolon would have come up a long time ago.

**Comment by vikas-nexcom** [comment]
@stevepiercy  Thanks for the reply. I really thought that as well. Absolutely, I can share the details:

Here is the request that is sent by angular after click on submit (from the chrome console) :

```
(id%3A+AGT)&transaction_date=2021-06-19+04%3A01%3A24&transaction_date-visible=2021-06-
19+04%3A01%3A24&monitoring_date=2021-06-19+04%3A01%3A24&monitoring_date-visible=2021-06-
19+04%3A01%3A24&transaction_comment=&__start__=recordings%3Amapping&__start__=%3Asequence&__end__=%3Ase
quence&__end__=recordings%3Amapping&token=fbc169099f65eb9ad1298a465a9955c833df714f&score_fail_color=danger&o
pt-Evaluation+Type=&opt-Call+Identification+Number=5584825&tags=&hidden-tags=&yesno-570553=true&force_comment-
570553=False&top_down_score-570553=True&max_score-570553=0&level-570553=1&is_leaf-570553=True&err_type-
570553=Business+Critical+Error&scorable-570553=True&known-570553=False&controllable-570553=True&process-
570553=False&did-570553=570553&mid-570553=&comm-570553=&yesno-570554=true&force_comment-
570554=False&top_down_score-570554=True&max_score-570554=0&level-570554=1&is_leaf-570554=True&err_type-
570554=End+User+Critical+Error&scorable-570554=True&known-570554=False&controllable-570554=True&process-
570554=False&did-570554=570554&mid-570554=&comm-570554=&biq_comment-161=&bia_comment-1435=&bia_comment-
1436=&bia_comment-1437=&biq_comment-162=&bia_comment-1438=&bia_comment-1439=&bia_comment-
1440=&bia_comment-1441=&bia_comment-1442=&bia_comment-1443=&bia_comment-1444=&saving=false&feedback_data=
[{"id":0,"mid":"","userId":"user","fullName":"","entryType":"something","body":"this;","isComplete":false,"inPerson":false,"planType"
:"Challenge","cAtId":"","courseName":"","datetimeCreated":"6/19/2021, 11:01 
AM","dateTimeModified":"","dueDate":"","userTz":"","canEdit":true,"transactionId":0,"programId":0,"feedbackReplies":[]}]
```

Here is the `request.body`  as  received by pyramid (looks identical to me):
 ```
 b'monitoring_start=2021-06-19+04%3A01%3A25-
05%3A00&_charset_=&__formid__=deform&islikert=None&formid=3716&uid=One%2C+Agent+
(id%3A+AGT)&transaction_date=2021-06-19+04%3A01%3A24&transaction_date-visible=2021-06-
19+04%3A01%3A24&monitoring_date=2021-06-19+04%3A01%3A24&monitoring_date-visible=2021-06-
19+04%3A01%3A24&transaction_comment=&__start__=recordings%3Amapping&__start__=%3Asequence&__end__=%3Ase
quence&__end__=recordings%3Amapping&token=fbc169099f65eb9ad1298a465a9955c833df714f&score_fail_color=danger&o
pt-Evaluation+Type=&opt-Call+Identification+Number=5584825&tags=&hidden-tags=&yesno-570553=true&force_comment-
570553=False&top_down_score-570553=True&max_score-570553=0&level-570553=1&is_leaf-570553=True&err_type-
570553=Business+Critical+Error&scorable-570553=True&known-570553=False&controllable-570553=True&process-
570553=False&did-570553=570553&mid-570553=&comm-570553=&yesno-570554=true&force_comment-
570554=False&top_down_score-570554=True&max_score-570554=0&level-570554=1&is_leaf-570554=True&err_type-
570554=End+User+Critical+Error&scorable-570554=True&known-570554=False&controllable-570554=True&process-
570554=False&did-570554=570554&mid-570554=&comm-570554=&biq_comment-161=&bia_comment-1435=&bia_comment-
1436=&bia_comment-1437=&biq_comment-162=&bia_comment-1438=&bia_comment-1439=&bia_comment-
1440=&bia_comment-1441=&bia_comment-1442=&bia_comment-1443=&bia_comment-1444=&saving=false&feedback_data=
[{"id":0,"mid":"","userId":"user","fullName":"","entryType":"something","body":"this;","isComplete":false,"inPerson":false,"planType
":"Challenge","cAtId":"","courseName":"","datetimeCreated":"6/19/2021, 11:01 
AM","dateTimeModified":"","dueDate":"","userTz":"","canEdit":true,"transactionId":0,"programId":0,"feedbackReplies":[]}]'

 ```
 
 Here is how `request.params` looks like upon printing (notice the `body` next to `entryType`:
 ```
NestedMultiDict([('monitoring_start', '2021-06-19 04:01:25-05:00'), ('_charset_', ''), ('__formid__', 'deform'), ('islikert', 'None'), 
('formid', '3716'), ('uid', 'One, Agent (id: AGT)'), ('transaction_date', '2021-06-19 04:01:24'), ('transaction_date-visible', '2021-06-
19 04:01:24'), ('monitoring_date', '2021-06-19 04:01:24'), ('monitoring_date-visible', '2021-06-19 04:01:24'), 
('transaction_comment', ''), ('__start__', 'recordings:mapping'), ('__start__', ':sequence'), ('__end__', ':sequence'), ('__end__', 
'recordings:mapping'), ('token', 'fbc169099f65eb9ad1298a465a9955c833df714f'), ('score_fail_color', 'danger'), ('opt-Evaluation 
Type', ''), ('opt-Call Identification Number', '5584825'), ('tags', ''), ('hidden-tags', ''), ('yesno-570553', 'true'), ('force_comment-
570553', 'False'), ('top_down_score-570553', 'True'), ('max_score-570553', '0'), ('level-570553', '1'), ('is_leaf-570553', 'True'), 
('err_type-570553', 'Business Critical Error'), ('scorable-570553', 'True'), ('known-570553', 'False'), ('controllable-570553', 'True'), 
('process-570553', 'False'), ('did-570553', '570553'), ('mid-570553', ''), ('comm-570553', ''), ('yesno-570554', 'true'), 
('force_comment-570554', 'False'), ('top_down_score-570554', 'True'), ('max_score-570554', '0'), ('level-570554', '1'), ('is_leaf-
570554', 'True'), ('err_type-570554', 'End User Critical Error'), ('scorable-570554', 'True'), ('known-570554', 'False'), 
('controllable-570554', 'True'), ('process-570554', 'False'), ('did-570554', '570554'), ('mid-570554', ''), ('comm-570554', ''), 
('biq_comment-161', ''), ('bia_comment-1435', ''), ('bia_comment-1436', ''), ('bia_comment-1437', ''), ('biq_comment-162', ''), 
('bia_comment-1438', ''), ('bia_comment-1439', ''), ('bia_comment-1440', ''), ('bia_comment-1441', ''), ('bia_comment-1442', ''), 
('bia_comment-1443', ''), ('bia_comment-1444', ''), ('saving', 'false'), ('feedback_data', 
'[{"id":0,"mid":"","userId":"user","fullName":"","entryType":"something","body":"this'), 
('","isComplete":false,"inPerson":false,"planType":"Challenge","cAtId":"","courseName":"","datetimeCreated":"6/1
9/2021, 11:01AM","dateTimeModified":"","dueDate":"","userTz":"","canEdit":true,"transactionId":0,"programId":0,"feedbackReplies":[]}]', '')])
 ```
 
 Additionally, here is how the `request.params` look after I remove the semicolon:
 
 ```
 NestedMultiDict([('monitoring_start', '2021-06-19 04:01:25-05:00'), ('_charset_', ''), ('__formid__', 'deform'), ('islikert', 'None'), 
('formid', '3716'), ('uid', 'One, Agent (id: AGT)'), ('transaction_date', '2021-06-19 04:01:24'), ('transaction_date-visible', '2021-06-
19 04:01:24'), ('monitoring_date', '2021-06-19 04:01:24'), ('monitoring_date-visible', '2021-06-19 04:01:24'), 
('transaction_comment', ''), ('__start__', 'recordings:mapping'), ('__start__', ':sequence'), ('__end__', ':sequence'), ('__end__', 
'recordings:mapping'), ('token', 'fbc169099f65eb9ad1298a465a9955c833df714f'), ('score_fail_color', 'danger'), ('opt-Evaluation 
Type', ''), ('opt-Call Identification Number', '5584825'), ('tags', ''), ('hidden-tags', ''), ('yesno-570553', 'true'), ('force_comment-
570553', 'False'), ('top_down_score-570553', 'True'), ('max_score-570553', '0'), ('level-570553', '1'), ('is_leaf-570553', 'True'), 
('err_type-570553', 'Business Critical Error'), ('scorable-570553', 'True'), ('known-570553', 'False'), ('controllable-570553', 'True'), 
('process-570553', 'False'), ('did-570553', '570553'), ('mid-570553', ''), ('comm-570553', ''), ('yesno-570554', 'true'), 
('force_comment-570554', 'False'), ('top_down_score-570554', 'True'), ('max_score-570554', '0'), ('level-570554', '1'), ('is_leaf-
570554', 'True'), ('err_type-570554', 'End User Critical Error'), ('scorable-570554', 'True'), ('known-570554', 'False'), 
('controllable-570554', 'True'), ('process-570554', 'False'), ('did-570554', '570554'), ('mid-570554', ''), ('comm-570554', ''), 
('biq_comment-161', ''), ('bia_comment-1435', ''), ('bia_comment-1436', ''), ('bia_comment-1437', ''), ('biq_comment-162', ''), 
('bia_comment-1438', ''), ('bia_comment-1439', ''), ('bia_comment-1440', ''), ('bia_comment-1441', ''), ('bia_comment-1442', ''), 
('bia_comment-1443', ''), ('bia_comment-1444', ''), ('saving', 'false'), ('feedback_data', 
'[{"id":-1,"cid":"","userId":"user","fullName":"","entryType":"something","body":"this","isComplete":false,"inPerson":false,"planType":"Challenge","cAtId":"","courseName":"","datetimeCreated":"6/19/2021, 11:01 
AM","dateTimeModified":"","dueDate":"","userTz":"","canEdit":true,"transactionId":0,"programId":0,"feedbackReplies":[]}]')])
```
 
 Let me know if this was what you were looking for.

**Comment by stevepiercy** [comment]
This is helpful, but I think we need something a little more complete. In the browser's web developer tools, click the tab for Network (on Chrome, Firefox might have a different name). Filter for XHR requests. This will allow you to get the raw headers, request params, and request body that Angular sends from the browser. With that...

How does Angular send the request, POST, GET, or other HTTP request?

What value do you have for the `Content-Type` header when you send the request?

If it is GET, then there might be an encoding issue. `;` is a reserved character in URLs. But I doubt that is the cause.

I also note that you are using Deform (and I assume Peppercorn) in your stack. It's probably not relevant, but just noting it.

**Comment by vikas-nexcom** [comment]
Thanks. It is a POST request. Here is the header output from XHR (**with a semicolon**):

```
Request URL: http://127.0.0.1:6543/mon/monitoring/552
Request Method: POST
Status Code: 500 Internal Server Error
Remote Address: 127.0.0.1:6543
Referrer Policy: strict-origin-when-cross-origin
Content-Length: 226
Content-Type: text/html; charset=UTF-8
Date: Sun, 20 Jun 2021 10:28:54 GMT
Server: Cheroot/8.5.2
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Connection: keep-alive
Content-Length: 1756
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Cookie: auth_tkt=54330876adb714886a9a5a4f457a28738118e8b18da6460fa22a70cba5d11db62d5a94f3b8a4e767760d65eb5de218cdbff91baacacb230263c3a460cf17ecQ09QQytnZW5h!userid_type:b64unicode;  mykey=63ff6d1207a27d7df3d4b2cb11c0095de5fc845d2b78c57bd3
Host: 127.0.0.1:6543
Origin: http://127.0.0.1:6543
Referer: http://127.0.0.1:6543/mon/monitoring/node
sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="90", "Google Chrome";v="90"
sec-ch-ua-mobile: ?0
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36
X-Requested-With: XMLHttpRequest
monitoring_start: 2021-06-20 05:27:41-05:00
_charset_: 
__formid__: deform
islikert: None
formid: 3716
uid: One, Agent (id: AGT)
transaction_date: 2021-06-20 05:27:40
transaction_date-visible: 2021-06-20 05:27:40
monitoring_date: 2021-06-20 05:27:40
monitoring_date-visible: 2021-06-20 05:27:40
transaction_comment: 
__start__: recordings:mapping
__start__: :sequence
__end__: :sequence
__end__: recordings:mapping
token: 36b1d1cb70d6b2e69f10408ef40319466648a39d
score_fail_color: danger
opt-Evaluation Type: 
opt-Call Identification Number: 5584825
tags: 
hidden-tags: 
yesno-570553: true
force_comment-570553: False
top_down_score-570553: True
max_score-570553: 0
level-570553: 1
is_leaf-570553: True
err_type-570553: Business Critical Error
scorable-570553: True
known-570553: False
controllable-570553: True
process-570553: False
did-570553: 570553
mid-570553: 
comm-570553: 
yesno-570554: true
force_comment-570554: False
top_down_score-570554: True
max_score-570554: 0
level-570554: 1
is_leaf-570554: True
err_type-570554: End User Critical Error
scorable-570554: True
known-570554: False
controllable-570554: True
process-570554: False
did-570554: 570554
mid-570554: 
comm-570554: 
biq_comment-161: 
bia_comment-1435: 
bia_comment-1436: 
bia_comment-1437: 
biq_comment-162: 
bia_comment-1438: 
bia_comment-1439: 
bia_comment-1440: 
bia_comment-1441: 
bia_comment-1442: 
bia_comment-1443: 
bia_comment-1444: 
saving: false
feedback_data: [{"id":0,"mid":"","userId":"user","fullName":"","entryType":"something","body":"this;","isComplete":false,"inPerson":false,"planType":"Challenge","cAtId":"","courseName":"","datetimeCreated":"6/20/2021, 12:28 PM","dateTimeModified":"","dueDate":"","userTz":"","canEdit":true,"transactionId":0,"programId":0,"feedbackReplies":[]}]
```

Here is header output (**without a semicolon**):

```
Request URL: http://127.0.0.1:6543/mon/monitoring/552
Request Method: POST
Status Code: 200 OK
Remote Address: 127.0.0.1:6543
Referrer Policy: strict-origin-when-cross-origin
Content-Length: 8
Content-Type: text/html; charset=UTF-8
Date: Sun, 20 Jun 2021 10:34:51 GMT
Server: Cheroot/8.5.2
Set-Cookie: auth_tkt=6f74da74f6bc528716d6d73cf2d9059c2546671b5212bdd2b9e15ba64df738db303dfd7dce8dd419462693510465bfd3080c570630e678ac8bba1c60cf19c3Q09Q!userid_type:b64unicode; Path=/; SameSite=Lax
Set-Cookie: auth_tkt=6f74da74f6bc528716d6d73cf2d9059c2546671b5212bdd2b9e15ba64df738db303dfd7dce8dd419462693510465bfd3080c570630e678ac8bba1c60cf19c3Q09Q!userid_type:b64unicode; Domain=127.0.0.1; Path=/; SameSite=Lax
Set-Cookie: auth_tkt=6f74da74f6bc528716d6d73cf2d9059c2546671b5212bdd2b9e15ba64df738db303dfd7dce8dd419462693510465bfd3080c570630e678ac8bba1c60cf19c3Q09QQytnZW5h!userid_type:b64unicode; Domain=.127.0.0.1; Path=/; SameSite=Lax
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Connection: keep-alive
Content-Length: 1756
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
Cookie: auth_tkt=4330876adb718d3d9a5a4f457a28738118e8b18da6460fa22a70cba5d11db62d5a94f3b8a4e767760d65eb5de218cdbff91baacacb230263c3a460cf17ecQ09QQytnZW5h!userid_type:b64unicode; mykey=3ff6d1207a38e26d7df3d4b2cb11c0095de5fc845d2b78c57bd3ca4e8ba
Host: 127.0.0.1:6543
Origin: http://127.0.0.1:6543
Referer: http://127.0.0.1:6543/mon/monitoring/node
sec-ch-ua: " Not A;Brand";v="99", "Chromium";v="90", "Google Chrome";v="90"
sec-ch-ua-mobile: ?0
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.212 Safari/537.36
X-Requested-With: XMLHttpRequest
monitoring_start: 2021-06-20 05:27:41-05:00
_charset_: 
__formid__: deform
islikert: None
formid: 3716
uid: One, Agent (id: AGT)
transaction_date: 2021-06-20 05:27:40
transaction_date-visible: 2021-06-20 05:27:40
monitoring_date: 2021-06-20 05:27:40
monitoring_date-visible: 2021-06-20 05:27:40
transaction_comment: 
__start__: recordings:mapping
__start__: :sequence
__end__: :sequence
__end__: recordings:mapping
token: 36b1d1cb70d6b2e69f10408ef40319466648
score_fail_color: danger
opt-Evaluation Type: 
opt-Call Identification Number: 55848
tags: 
hidden-tags: 
yesno-570553: true
force_comment-570553: False
top_down_score-570553: True
max_score-570553: 0
level-570553: 1
is_leaf-570553: True
err_type-570553: Business Critical Error
scorable-570553: True
known-570553: False
controllable-570553: True
process-570553: False
did-570553: 570553
mid-570553: 
comm-570553: 
yesno-570554: true
force_comment-570554: False
top_down_score-570554: True
max_score-570554: 0
level-570554: 1
is_leaf-570554: True
err_type-570554: End User Critical Error
scorable-570554: True
known-570554: False
controllable-570554: True
process-570554: False
did-570554: 570554
mid-570554: 
comm-570554: 
biq_comment-161: 
bia_comment-1435: 
bia_comment-1436: 
bia_comment-1437: 
biq_comment-162: 
bia_comment-1438: 
bia_comment-1439: 
bia_comment-1440: 
bia_comment-1441: 
bia_comment-1442: 
bia_comment-1443: 
bia_comment-1444: 
saving: false
feedback_data: [{"id":-1,"mid":"","userId":"user","fullName":"","entryType":"something","body":"this","isComplete":false,"inPerson":false,"planType":"Challenge","cAtId":"","courseName":"","datetimeCreated":"6/20/2021, 12:28 PM","dateTimeModified":"","dueDate":"","userTz":"","canEdit":true,"transactionId":0,"programId":0,"feedbackReplies":[]}]
```

I hope this helps. Yes, I see  `peppercorn==0.6` in the `requirements.txt` file.

**Comment by vikas-nexcom** [comment]
@stevepiercy  I discovered something on Angular's GitHub ( issue: https://github.com/angular/angular.js/issues/9224) today. I didn't know this before but the semicolon should be properly encoded by angular ( i.e. %3B)  and I think that's what pyramid excepts. If I wrote my text like (**this%3B** instead of **this;**). Everything works great. Pyramid even detect as "this;". It doesn't solve my issue fully but gives me a path to look into. Is there a quick way to fix the encoding in Python/Pyramid?

**Comment by stevepiercy** [comment]
This is question for @bertjwregeer or @mmerickel about where encoding should occur. I'm not entirely sure myself.

If it were my app, I would not send the JSON object as a parameter (`feedback_data`). I would send it as data in the request, like [requests](https://docs.python-requests.org/en/latest/api/) (see `json`) or [curl](https://curl.se/docs/manpage.html#-d) do. If I could not do that, then I would percent-encode each value in my JSON object. But Angular is JavaScript, not Python or a shell, but I assume Angular has something similar. I don't use it, so I don't know for sure. HTH.

**Comment by mmerickel** [comment]
`request.params` requires data in either application/x-www-form-urlencoded or multipart/form-data format. This data is not and thus there is no guarantee it will parse properly. The only question here is whether a `;` in the query string is ok in these specs or not - if it's not allowed then it needs to be escaped.

Following the specs from https://www.w3.org/TR/html401/interact/forms.html#h-17.13.4 you'll find:

> Control names and values are escaped. Space characters are replaced by `+', and then reserved characters are escaped as described in [RFC1738], section 2.2: Non-alphanumeric characters are replaced by `%HH', a percent sign and two hexadecimal digits representing the ASCII code of the character. Line breaks are represented as "CR LF" pairs (i.e., `%0D%0A').

Which references RFC1738 which states:

> Thus, only alphanumerics, the special characters "$-_.+!*'(),", and
>    reserved characters used for their reserved purposes may be used
>    unencoded within a URL.

So basically there isn't a bug here and angular should be escaping the `;` character.

**Comment by ztane** [comment]
@mmerickel I started googling around this and it turns out closing this might not be entirely correct either. The current *decoding* algorithm is https://www.w3.org/TR/2014/REC-html5-20141028/forms.html#url-encoded-form-data :

> Let strings be the result of **strictly** splitting the string payload on U+0026 AMPERSAND characters (&).

The word strictly is even emphasized there.

**Comment by ztane** [comment]
I believe the ; semicolon is a misunderstanding - it has been part of query strings but I believe it was **never ever used** in POST payloads. It is allowed for URLs exactly because if & is used then you always need to entity-escape them in HTML source making their use rather tedious in hand-written HTML.

**Comment by mmerickel** [comment]
@ztane thanks - you may be correct but I'm not convinced. I think at the very least this issue should be opened against Webob instead of Pyramid and someone needs to construct a failing test case that's easier to decipher. My concern is that the data contains quite a few characters that are documented to need to be escaped as I read things such as the `:`, `{`, and `[` in json as well. I just focused on the `;` because that's what people were discussing above. Do you really think it should be safe to send `foo=<json>` data through unescaped and count as valid application/x-www-form-urlencoded data? For this type of data I'm fairly certain webob is using the stdlib parse_qs function and there's little for us to do here other than claim we're misusing the function signature, or that the bug affects the stdlib.

**Comment by bertjwregeer** [comment]
WebOb doesn't do any parsing itself for form encoded data in a POST, it is all handled by `cgi.FieldStorage`.

I would indeed want a proper test that showcases the issue, the above information does not provide enough information for me to understand what data is being sent and where. Even a repository that reproduces the issue with the frontend/backend components would be great.

**Comment by bertjwregeer** [comment]
Also `request.params` mixes and matches `GET` and `POST`.
